### DISCUSSION TITLE
is queryParameters a real required field when configuring prometheus triggers?

### DISCUSSION THREAD

**Discussion by ahinh43** [kedacore/keda-docs#1391] [Q&A]
I'm working on upgrading our Keda deployment from 2.11 to 2.14, and one of the changes introduced in 2.13 is the [queryParameters field](https://github.com/kedacore/keda/issues/4962) for the prometheus scaler. However, it's documented as a required field yet in the [source code the field is marked optional](https://github.com/kedacore/keda/blob/main/pkg/scalers/prometheus_scaler.go#L58). 

My initial impression of reading the [trigger specifications](https://keda.sh/docs/2.14/scalers/prometheus/#trigger-specification)
is that missing fields under required fields would result in the request being rejected during an apply but I could be misunderstanding this.

Is there any implications for not setting queryParameters in a prometheus trigger's metadata field?

**Comment by zroubalik** [comment]
Good catch, this should be optional, could you please open a PR to fix this please?
